cse-runner-app/src/test/java/com/farao_community/farao/cse/runner/app/util/MerchantLineTest.java

```diff
@@ -50,11 +69,28 @@ void testTransformerSettings() {
 
     @Test
     void testWithLoadFlow() {
-        MerchantLine.activateMerchantLine(ProcessType.IDCC, network);
-
+        MerchantLine.activateMerchantLine(ProcessType.IDCC, network, cseData);
+        TwoWindingsTransformer twoWindingsTransformer = network.getTwoWindingsTransformer(MerchantLine.MENDRISIO_ID);
+        PhaseTapChanger phaseTapChanger = twoWindingsTransformer.getPhaseTapChanger();
         assertEquals(0, phaseTapChanger.getTapPosition());
         LoadFlow.run(network, LoadFlowParameters.load());
         assertEquals(12, phaseTapChanger.getTapPosition());
         assertEquals(-300, twoWindingsTransformer.getTerminal1().getP(), DOUBLE_PRECISION_FOR_REGULATED_FLOW);
     }
+
+    @Test
+    void testTransformerSettingsForD2ccProcess() {
+        Map<String, Double> fixedFlowLines = Map.of(
+                MerchantLine.MENDRISIO_CAGNO_CODE_IN_NTC_FILE, 75.
+        );
+        TwoWindingsTransformer twoWindingsTransformer = networkWithMendrisioCagnoLine.getTwoWindingsTransformer(MerchantLine.MENDRISIO_ID);
+        PhaseTapChanger phaseTapChanger = twoWindingsTransformer.getPhaseTapChanger();
+        Mockito.when(cseData.getNtc()).thenReturn(ntc);
+        Mockito.when(cseData.getNtc().getFlowOnFixedFlowLines()).thenReturn(fixedFlowLines);
+        Mockito.when(cseData.getLineFixedFlows()).thenReturn(lineFixedFlows);
+        when(cseData.getLineFixedFlows().getFixedFlow(Mockito.any(), Mockito.any(), Mockito.any())).thenReturn(Optional.empty());
+        MerchantLine.activateMerchantLine(ProcessType.D2CC, networkWithMendrisioCagnoLine, cseData);
+        assertEquals(95.58, phaseTapChanger.getRegulationValue(), DOUBLE_PRECISION);
```

Comment:
You must check flow on mendrisio cagno line which is the value we are looking for in the end. And make sure it get close to 75MW. Don't forget to run a loadflow in the test after merchant line activation to check this value.